ros_ws/src/car_control/include/drive_parameters_source.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <ros/ros.h>
+
+#include <drive_msgs/drive_param.h>
+#include <functional>
+#include <std_msgs/Float64.h>
+
+class DriveParametersSource;
+typedef std::function<void(DriveParametersSource*, const drive_msgs::drive_param::ConstPtr&)>
+    DriveParameterCallbackFunction;
+
+constexpr double IDLE_RANGE = 0.01;
+
+/*
+*  A class that listens on a topic that publishes drive parameters and stores information about that source
+*/
+class DriveParametersSource
+{
+    public:
+    /**
+     * @brief Construct a new DriveParametersSource object,
+     * subscribes to the topic and stores the parameters into local variables.
+     *
+     * @param node_handle A node handle is needed to create a topic subscriber
+     * @param topic The name of the drive_param topic to subscribe to
+     * @param update_callback Callback function that will be called whenever a message was received
+     * @param priority Priority of the source. Sources with higher priority will be preferred.
+     * @param timeout Messages will be deferred when they are older than this, in seconds.
+     */
+    DriveParametersSource(ros::NodeHandle* node_handle, const char* topic,
+                          DriveParameterCallbackFunction update_callback, int priority, double timeout);
+
+    /**
+     * @brief Returns true if no update was received for a certain time, determined by the timeout variable.
+     */
+    bool isOutdated();
+
+    /**
+     * @brief Returns true if the last update contained values close to 0 for both steering and throttle.
+     * If no update was received yet, this returns true.
+     */
+    bool isIdle();
+
+    int getPriority();
```

Comment:
Sometimes priority is defined inverse such that the highest priority is 0. To prevent any confusion, may add a comment clarifying whether highest priority means highest or lowest value.